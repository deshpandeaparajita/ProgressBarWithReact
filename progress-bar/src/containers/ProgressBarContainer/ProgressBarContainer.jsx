import React, { useEffect, useState } from 'react';
import IncrementButtonComponent from '../../components/IncrementButtonComponent/IncrementButtonComponent';
import { ProgressBarComponent } from '../../components/ProgressBarComponent';
import './ProgressBarContainer.css';
import { Container, Row, Col } from 'react-bootstrap';

export const ProgressBarContainer = (props) => {
    const [data, setData] = useState({});
    const [range, setRange] = useState([]);

    useEffect(() => {
        fetch("http://pb-api.herokuapp.com/bars").then(res => {
            if (res.ok) {
                res.json().then(resObj => {
                    setData(resObj);
                    setRange(resObj.bars);
                });
            } else console.log("unable to fetch");
        });

    }, []);

    const { buttons, limit } = data;

    const handleClick = (e, value) => {
        e.preventDefault();

        let newRange = [];
        range && range.map((val) => {

            if (val + value < 0) {
                newRange.push(0);
            } else if (val + value > limit) {
                newRange.push(limit);
            } else {
                newRange.push(val + value);
            }
        })
        setRange(newRange);

    }

    function getSum(total, num) {
        return total + num;
    }

    console.log("range: ", range);
    return (
        <Container className="main">
            <Row style={{ marginBottom: "20px" }} >
                <Col>
                    Total usage : {range.reduce(getSum, 0)}
                </Col>
            </Row>
            <Row >
                <Col>
                    {range && range.map((value) => {
                        return (
                            <ProgressBarComponent percent={value} limit={limit} />
                        );
                    })
                    }
                </Col>
            </Row>
            <Row >
                <Col>
                    <div>
                        <IncrementButtonComponent buttons={buttons} handleClick={(e, value) => handleClick(e, value)} />
                    </div>
                </Col>
            </Row>

        </Container >
    );
}

export default ProgressBarContainer;