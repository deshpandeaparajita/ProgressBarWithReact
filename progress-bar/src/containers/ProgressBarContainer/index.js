export {default as ProgressBarContainer} from './ProgressBarContainer';