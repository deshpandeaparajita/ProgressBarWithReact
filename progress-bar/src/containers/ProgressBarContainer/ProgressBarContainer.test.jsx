import React from 'react';
import { shallow } from 'enzyme';
import ProgressBarContainer from './ProgressBarContainer';



describe('ProgressBarContainer', () => {
    const props = {
        handleClick: jest.fn(),
    };

    it('should match snapshot', () => {
        const component = shallow(<ProgressBarContainer {... props} />);
        expect(component).toMatchSnapshot();
    });
});