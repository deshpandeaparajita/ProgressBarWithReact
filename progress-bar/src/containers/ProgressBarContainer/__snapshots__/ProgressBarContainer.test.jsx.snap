// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBarContainer should match snapshot 1`] = `ShallowWrapper {}`;

exports[`ProgressBarContainer should match snapshot when click on increment button 1`] = `ShallowWrapper {}`;
